File: MiniDOMOffice/Magazzino/Inventario/Collegamenti/Collection/OggettoCollegatiCollection.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Store;


namespace minidom
{
    public partial class Store
    {

        /// <summary>
        /// Collezione di oggetti collegati ad un oggetto
        /// </summary>
        [Serializable]
        public class OggettiCollegatiCollection 
            : CCollection<OggettoCollegato>
        {
            [NonSerialized]
            private OggettoInventariato m_Owner;

            /// <summary>
            /// Costruttore
            /// </summary>
            public OggettiCollegatiCollection()
            {
                m_Owner = null;
            }

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="owner"></param>
            public OggettiCollegatiCollection(OggettoInventariato owner) : this()
            {
                Load(owner);
            }

            /// <summary>
            /// Imposta l'oggetto
            /// </summary>
            /// <param name="owner"></param>
            protected internal virtual void SetOwner(OggettoInventariato owner)
            {
                m_Owner = owner;
                //if (owner is object)
                //{
                //    foreach(var c in this)
                //    {
                //        c.SetOggetto1(owner);
                //    }
                //}
            }

            /// <summary>
            /// Carica le relazioni
            /// </summary>
            /// <param name="owner"></param>
            protected internal virtual void Load(OggettoInventariato owner)
            {
                if (owner is null)
                    throw new ArgumentNullException("owner");
                Clear();
                m_Owner = owner;
                int oID = DBUtils.GetID(owner, 0);
                if (oID == 0)
                    return;
                using (var cursor = new OggettoCollegatoCursor())
                {
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    //cursor.IDOggetto1.Value = DBUtils.GetID(owner, 0);
                    cursor.WhereClauses *= (cursor.Field("IDOggetto1").EQ(oID) + cursor.Field("IDOggetto2").EQ(oID));
                    while (cursor.Read())
                    {
                        var item = cursor.Item;
                        if (item.IDOggetto1 == oID)
                            item.SetOggetto1(owner);
                        if (item.IDOggetto2 == oID)
                            item.SetOggetto2(owner);
                        this.Add(item);                        
                    }
                }
            }
        }
    }
}

File: MiniDOMOffice/Magazzino/Inventario/Collegamenti/OggettoCollegatiCollection.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Office

    <Serializable>
    Public Class OggettiCollegatiCollection
        Inherits CCollection(Of OggettoCollegato)

        <NonSerialized> Private m_Owner As OggettoInventariato

        Public Sub New()
            Me.m_Owner = Nothing
        End Sub

        Public Sub New(ByVal owner As OggettoInventariato)
            Me.New()
            Me.Load(owner)
        End Sub

        Protected Friend Overridable Sub SetOwner(ByVal owner As OggettoInventariato)
            Me.m_Owner = owner
        End Sub
 
         
        Protected Friend Overridable Sub Load(ByVal owner As OggettoInventariato)
            If (owner Is Nothing) Then Throw New ArgumentNullException("owner")
            Me.Clear()
            Me.m_Owner = owner
            Dim oID As Integer = GetID(owner)
            If (oID = 0) Then Exit Sub
            Dim dbSQL As String = "SELECT * FROM [tbl_OfficeOggettiCollegati] WHERE [Stato]=" & ObjectStatus.OBJECT_VALID & " AND ([IDOggetto1]=" & oID & " OR [IDOggetto2]=" & oID & ")"
            Dim dbRis As System.Data.IDataReader = Office.Database.ExecuteReader(dbSQL)
            While dbRis.Read
                Dim item As New OggettoCollegato
                Office.Database.Load(item, dbRis)
                Me.Add(item)
                If (item.IDOggetto1 = oID) Then item.SetOggetto1(owner)
                If (item.IDOggetto2 = oID) Then item.SetOggetto2(owner)
            End While
            dbRis.Dispose()
        End Sub

    End Class

End Class


